dipy/nn/synb0.py

```diff
@@ -0,0 +1,410 @@
+#!/usr/bin/python
+"""
+Class and helper functions for fitting the Synb0 model.
+"""
+
+
+from packaging.version import Version
+import logging
+from dipy.data import get_fnames
+from dipy.testing.decorators import doctest_skip_parser
+from dipy.utils.optpkg import optional_package
+import numpy as np
+
+tf, have_tf, _ = optional_package('tensorflow')
+tfa, have_tfa, _ = optional_package('tensorflow_addons')
+if have_tf and have_tfa:
+    from tensorflow.keras.models import Model
+    from tensorflow.keras.layers import MaxPool3D, Conv3DTranspose, Conv3D, LeakyReLU, Concatenate, Layer
+    from tensorflow_addons.layers import InstanceNormalization
+    if Version(tf.__version__) < Version('2.0.0'):
+        raise ImportError('Please upgrade to TensorFlow 2+')
+
+logging.basicConfig()
+logger = logging.getLogger('synb0')
+
+
+def set_logger_level(log_level):
+    """ Change the logger of the Synb0 to one on the following:
+    DEBUG, INFO, WARNING, CRITICAL, ERROR
+
+    Parameters
+    ----------
+    log_level : str
+        Log level for the Synb0 only
+    """
+    logger.setLevel(level=log_level)
+
+
+class Synb0():
+    """
+    This class is intended for the Synb0 model.
+    The model is the deep learning part of the Synb0-Disco
+    pipeline, thus stand-alone usage is not
+    recommended.
+    """
+
+    @doctest_skip_parser
+    def __init__(self, verbose=False):
+        r"""
+        The model was pre-trained for usage on pre-processed images
+        following the synb0-disco pipeline.
+        One can load their own weights using load_model_weights.
+
+        This model is designed to take as input
+        a b0 image and a T1 weighted image.
+
+        It was designed to predict a b-inf image.
+
+        Parameters
+        ----------
+        verbose : bool (optional)
+            Whether to show information about the processing.
+            Default: False
+
+        References
+        ----------
+        ..  [1] Schilling, K. G., Blaber, J., Huo, Y., Newton, A.,
+            Hansen, C., Nath, V., ... & Landman, B. A. (2019).
+            Synthesized b0 for diffusion distortion correction (Synb0-DisCo).
+            Magnetic resonance imaging, 64, 62-70.
+        ..  [2] Schilling, K. G., Blaber, J., Hansen, C., Cai, L.,
+            Rogers, B., Anderson, A. W., ... & Landman, B. A. (2020).
+            Distortion correction of diffusion weighted MRI without reverse
+            phase-encoding scans or field-maps.
+            PloS one, 15(7), e0236418.
+        """
+
+        if not have_tf:
+            raise tf()
+        if not have_tfa:
+            raise tfa()
+
+        log_level = 'INFO' if verbose else 'CRITICAL'
+        set_logger_level(log_level)
+
+        # Synb0 network load
+
+        self.model = self.UNet3D()
+        self.model.build(input_shape=(None, 80, 80, 96, 2))
+    
+    class UNet3D(Model):
```

Comment:
Hi @pjsjongsung,

When TensorFlow is not present,  `Model` is not imported, so this generates an error because it does not know what it is.....

Actually, it is not a test which is failing but when pytest is parsing all the files to find tests

Reply:
BTW, in resdnn, they never inherit from a model, that is the difference.

Reply:
Is this line something I can add to above the class definition to solve this?
`@doctest_skip_parser`

Reply:
I see. I'll make changes so that it doesn't do that then. Thank you!